src/core/ImportNode.cc

```diff
@@ -163,37 +192,37 @@ std::unique_ptr<const Geometry> ImportNode::createGeometry() const
 
   switch (this->type) {
   case ImportType::STL: {
-    g = import_stl(this->filename, loc);
+    g = optionally_center(import_stl(this->filename, loc), this->center);
     break;
   }
   case ImportType::AMF: {
-    g = import_amf(this->filename, loc);
+    g = optionally_center(import_amf(this->filename, loc), this->center);
     break;
   }
   case ImportType::_3MF: {
-    g = import_3mf(this->filename, loc);
+    g = optionally_center(import_3mf(this->filename, loc), this->center);
     break;
   }
   case ImportType::OFF: {
-    g = import_off(this->filename, loc);
+    g = optionally_center(import_off(this->filename, loc), this->center);
     break;
   }
   case ImportType::OBJ: {
-    g = import_obj(this->filename, loc);
+    g = optionally_center(import_obj(this->filename, loc), this->center);
     break;
   }
   case ImportType::SVG: {
-    g = import_svg(this->fn, this->fs, this->fa, this->filename, this->id, this->layer, this->dpi, this->center, loc);
+    g = optionally_center(import_svg(this->fn, this->fs, this->fa, this->filename, this->id, this->layer, this->dpi, false, loc), this->center);
     break;
   }
   case ImportType::DXF: {
     DxfData dd(this->fn, this->fs, this->fa, this->filename, this->layer.value_or(""), this->origin_x, this->origin_y, this->scale);
-    g = dd.toPolygon2d();
+    g = optionally_center(dd.toPolygon2d(), this->center);
     break;
   }
 #ifdef ENABLE_CGAL
   case ImportType::NEF3: {
-    g = import_nef3(this->filename, loc);
+    g = optionally_center(import_nef3(this->filename, loc), this->center);
```

Comment:
Since all the cases do exactly the same: Combined with my earlier suggestion, you could pull optionally_center() out of the switch and just do `if (this->center) g = center(g)`, or possibly also move center() into the Geometry class.

Reply:
I wasn't sure if centering should stay in the import or become generally available.

I would love to implement it on Geometry, but the issue I was having is that the 2D transform on Polygon2d can't be called once it becomes a Geometry. I could implement `center_on_origin()` as an abstract Geometry method for each of PolySet, Polygon2d and  CGAL_Nef_polyhedron, but it would be nice if I could only do it on Geometry alone.

Reply:
Yeah, it follows the other case. Let's leave it like this, and if we refactor Geometry in the future, code can be unified a bit..

Reply:
well, I could do a 3D implementation in Geometry, and then override it in Polygon2d. not really any more duplication than there is now. would that be preferrable?

Reply:
Nah, let's just leave it for now. We're considering allowing 3D transformations on 2D geometry, in which case we need to rethink some of the APIs, so we can defer all this until after that's sorted.